### PR TITLE
feat: List Page

### PR DESCRIPTION
**Pull request by cbachmeier**
- Layout changes for List Page
- Moves ListingGrid to its own file
- Mobile support

https://uniswaplabs.atlassian.net/browse/WEB-2357
Depends on #5314

### CODE REVIEW

**Comment by vm** [comment]
Once the dropdown is merged and this is rebased I can start on the review 👍

**Comment by cbachmeier** [comment]
I also just want to add the context that this is an interim solution for launch and the more polished page will be built after

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+
+  useEffect(() => {
+    inputRef.current.value = listPrice !== undefined ? `${listPrice}` : ''
+    setWarningType(WarningType.NONE)
+    if (!warning && listPrice) {
+      if (listPrice < asset.floorPrice) setWarningType(WarningType.BELOW_FLOOR)
+      else if (asset.floor_sell_order_price && listPrice >= asset.floor_sell_order_price)
+        setWarningType(WarningType.ALREADY_LISTED)
+    } else if (warning && listPrice && listPrice >= 0) removeMarketplaceWarning(asset, warning)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:146]: likely don't need this?

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+  setGlobalOverride,
+  globalOverride,
+  warning,
+  asset,
+}: PriceTextInputProps) => {
+  const [focused, setFocused] = useState(false)
+  const [warningType, setWarningType] = useState(WarningType.NONE)
+  const removeMarketplaceWarning = useSellAsset((state) => state.removeMarketplaceWarning)
+  const removeSellAsset = useSellAsset((state) => state.removeSellAsset)
+  const inputRef = useRef() as React.MutableRefObject<HTMLInputElement>
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:136]: is there a way to do this without a cast?

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+  listPrice?: number
+  setListPrice: Dispatch<number | undefined>
+  isGlobalPrice: boolean
+  setGlobalOverride: Dispatch<boolean>
+  globalOverride: boolean
+  warning?: ListingWarning
+  asset: WalletAsset
+}
+
+const PriceTextInput = ({
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:123]: nit: i would put this in its own file to simplify the existing file

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+
+const NFTListRow = ({ asset, globalPriceMethod, globalPrice, setGlobalPrice, selectedMarkets }: NFTListRowProps) => {
+  const [expandMarketplaceRows, setExpandMarketplaceRows] = useState(false)
+  const removeAsset = useSellAsset((state) => state.removeSellAsset)
+  const [localMarkets, setLocalMarkets] = useState([])
+  const [hovered, setHovered] = useState(false)
+  const handleHover = () => setHovered(!hovered)
+
+  useEffect(() => {
+    setLocalMarkets(JSON.parse(JSON.stringify(selectedMarkets)))
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:469]: why this stringify and parse?

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+export interface NFTListRowProps {
+  asset: WalletAsset
+  globalPriceMethod?: SetPriceMethod
+  setGlobalPrice: Dispatch<number | undefined>
+  globalPrice?: number
+  selectedMarkets: ListingMarket[]
+}
+
+const NFTListRow = ({ asset, globalPriceMethod, globalPrice, setGlobalPrice, selectedMarkets }: NFTListRowProps) => {
+  const [expandMarketplaceRows, setExpandMarketplaceRows] = useState(false)
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:462]: do you need this separate piece of state. could you just have `expandMarketplaceRows = marketplaces.length >= 2`?

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,550 @@
...
+  globalPriceMethod?: SetPriceMethod
+  globalPrice?: number
+  setGlobalPrice: Dispatch<number | undefined>
+  selectedMarkets: ListingMarket[]
+  removeMarket?: () => void
+  asset: WalletAsset
+  showMarketplaceLogo: boolean
+}
+
+const MarketplaceRow = ({
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:292]: ideally this also would be a diff file

File: src/nft/components/profile/list/ListPage.tsx [hunk 7]
```diff
@@ -1,583 +1,59 @@
...
 
-const NFTListRow = ({ asset, globalPriceMethod, globalPrice, setGlobalPrice, selectedMarkets }: NFTListRowProps) => {
-  const [expandMarketplaceRows, setExpandMarketplaceRows] = useState(false)
-  const removeAsset = useSellAsset((state) => state.removeSellAsset)
-  const [localMarkets, setLocalMarkets] = useState([])
-  const [hovered, setHovered] = useState(false)
-  const handleHover = () => setHovered(!hovered)
+const GridWrapper = styled.div`
+  margin-top: 24px;
+  margin-left: 0px;
```
Comment (vm) [inline src/nft/components/profile/list/ListPage.tsx:43]: nit: don't need this

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 8]
```diff
@@ -0,0 +1,550 @@
...
+      </Row>
+      <Row style={{ flex: '1.5' }} display={{ sm: 'none', md: 'flex' }}>
+        <Column width="full">
+          <EthPriceDisplay ethPrice={userReceives} />
+          {(showGlobalPrice ? globalPrice && globalPrice !== 0 : listPrice !== 0) && (
+            <Row marginTop="4" width="full" fontSize="12" color="textSecondary">
+              {profit ? <Box marginLeft="auto">Profit: {formatEth(profit)} ETH</Box> : null}
+              {profitPercent ? (
+                <Box marginLeft="8" marginRight="0">
+                  ({profitPercent > 0 && '+'}
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:441]: dont need -?

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 9]
```diff
@@ -0,0 +1,550 @@
...
+      </Row>
+      <Row flex="1" display={{ sm: 'none', md: 'flex' }}>
+        <Box className={body} color="textSecondary" width="full" textAlign="right">
+          {(asset.basisPoints * 0.01).toFixed(1)}%
+        </Box>
+      </Row>
+      <Row style={{ flex: '1.5' }} display={{ sm: 'none', md: 'flex' }}>
+        <Column width="full">
+          <EthPriceDisplay ethPrice={userReceives} />
+          {(showGlobalPrice ? globalPrice && globalPrice !== 0 : listPrice !== 0) && (
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:436]: could you make this a variable? ideally we would do if blocks bc its sorta hard to follow. some comments also would be helpful here

File: src/nft/components/profile/list/NFTListingsGrid.tsx [hunk 10]
```diff
@@ -0,0 +1,550 @@
...
+  globalPriceMethod,
+  globalPrice,
+  setGlobalPrice,
+  selectedMarkets,
+  removeMarket = undefined,
+  asset,
+  showMarketplaceLogo,
+}: MarketplaceRowProps) => {
+  const [listPrice, setListPrice] = useState<number>()
+  const [globalOverride, setGlobalOverride] = useState(false)
```
Comment (vm) [inline src/nft/components/profile/list/NFTListingsGrid.tsx:302]: could you explain what this override is?

File: src/nft/components/profile/list/NFTListingsGrid.tsx (hunk 5 above)
Comment (cbachmeier) [reply src/nft/components/profile/list/NFTListingsGrid.tsx:462]: a listing can be for multiple marketplaces, so this is to toggle whether a user is splitting the row to view on a per marketplace basis or for generally across all

**Comment by cbachmeier** [comment]
All the code in src/nft/components/profile/list/NFTListingsGrid.tsx was just copied from what's current into it's own file, this will all be changed when we move to the polished designs

**Review by vm — APPROVED** [review]
